Dockerfile

```diff
@@ -652,7 +652,7 @@ COPY --chmod=0644 hack/udevd/80-net-name-slot.rules /rootfs/usr/lib/udev/rules.d
 COPY --chmod=0644 hack/lvm.conf /rootfs/etc/lvm/lvm.conf
 RUN <<END
     ln -s /usr/share/zoneinfo/Etc/UTC /rootfs/etc/localtime
-    touch /rootfs/etc/{extensions.yaml,resolv.conf,hosts,os-release,machine-id,cri/conf.d/cri.toml,cri/conf.d/01-registries.part,cri/conf.d/20-customization.part}
+    touch /rootfs/etc/{extensions.yaml,resolv.conf,resolv-global.conf,hosts,os-release,machine-id,cri/conf.d/cri.toml,cri/conf.d/01-registries.part,cri/conf.d/20-customization.part}
```

Comment:
I wonder if this is a good name, it doesn't have to be in `/etc` at all, we could perfectly write it to `tmpfs` under `/system`. This way we don't have to manage it with bind-mount tricks.

The `/etc/resolv.conf` is there as it's a well known path, while the path to the `resolv.conf` we pass to the `kubelet` doesn't have to be anything fixed.

Reply:
That is possible, but we will have to rework `EtcFileController` or/and `EtcFileController`.

Reply:
There's totally no need to do that, just write down contents to e.g. `/system/resolved/resolv.conf` for example. It's tmpfs, always mounted and writeable, you don't need any kind of `EtcFileController` there

Reply:
There is `renderResolvConf` method in `EtcFileController` and it also contains dns resolve logic other than just simple write of `nameserver 127.0.0.1`. We will have to copy/modify it.

Reply:
Oh, okay. Yeah now `github.com/siderolabs/talos/internal/app/machined/pkg/controllers/network.EtcFileController` writes directly to `/system/resolved/resolv.conf`.

> manage it with bind-mount tricks.

It's still is in `github.com/siderolabs/talos/internal/app/machined/pkg/system/services.Kubelet.Runner` otherwise it doesn't work.

Reply:
yes, I mean bind-mount of different sort when we write to read-only `/etc` ;)